jenkins/automation/rgw/ceph/cortx-rgw-build-container-image.groovy

```diff
@@ -0,0 +1,137 @@
+        string(name: 'CEPH_CONTAINER_BRANCH', defaultValue: 'rgw-rockylinux-test', description: 'Branch or GitHash for ceph container image scripts.', trim: true)
+        string(name: 'CEPH_RELEASE', defaultValue: 'quincy', description: 'Ceph release to build image from.', trim: true)
+        string(name: 'OS_IMAGE', defaultValue: 'rockylinux', description: 'Base OS docker image to build from.', trim: true)
+        string(name: 'OS_IMAGE_TAG', defaultValue: '8', description: 'OS docker image tag.', trim: true)
+    }    
+
+    stages {
+        stage ('Checkout ceph-container repo') {
+            steps { 
+                cleanWs()            
+                script {
+                    checkout([$class: 'GitSCM', branches: [[name: "${CEPH_CONTAINER_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${CEPH_CONTAINER_REPO}"]]])                
+                }
+            }
+        }
+
+        stage ('Build Ceph Container Image') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: 'Build Ceph Container Image', script: """
+                    echo "Starting Build"
+                    make FLAVORS=${CEPH_RELEASE},${OS_IMAGE},${OS_IMAGE_TAG} RELEASE=${CEPH_CONTAINER_BRANCH} build
+
+                    echo -e "==============================\n"
+
+                    echo "List created images:"
+                    # docker images --format "{{.Repository}}:{{.Tag}}" --filter reference=${CEPH_PROJECT}/daemon-base:${CEPH_CONTAINER_BRANCH}-${CEPH_RELEASE}-${OS_IMAGE}-${OS_IMAGE_TAG}-${ARCH}
+                    # docker images --format "{{.Repository}}:{{.Tag}}" --filter reference=${CEPH_PROJECT}/daemon:${CEPH_CONTAINER_BRANCH}-${CEPH_RELEASE}-${OS_IMAGE}-${OS_IMAGE_TAG}-${ARCH}
+                    docker images --format "{{.Repository}}:{{.Tag}}" --filter reference=ceph/daemon-base:${CEPH_CONTAINER_BRANCH}-${CEPH_RELEASE}-${OS_IMAGE}-${OS_IMAGE_TAG}-${ARCH}
+                    docker images --format "{{.Repository}}:{{.Tag}}" --filter reference=ceph/daemon:${CEPH_CONTAINER_BRANCH}-${CEPH_RELEASE}-${OS_IMAGE}-${OS_IMAGE_TAG}-${ARCH}
+
+                """
+            }
+        }
+
+        stage ('Push Image to Registry') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: 'Push Image to Registry', script: """
+
+                    echo "Tag container images"
+                    # docker tag ${CEPH_PROJECT}/daemon:${CEPH_CONTAINER_BRANCH}-${CEPH_RELEASE}-${OS_IMAGE}-${OS_IMAGE_TAG}-${ARCH} ${REGISTRY}/${CEPH_PROJECT}/${CEPH_RELEASE}-${OS_IMAGE}_${OS_IMAGE_TAG}:daemon-${CEPH_CONTAINER_BRANCH}-${CEPH_RELEASE}-${OS_IMAGE}_${OS_IMAGE_TAG}-${ARCH}-build_${BUILD_NUMBER}
```

Comment:
remove commented code if not required

Reply:
This is a test job for future reference. Currently it will be disabled. All comments are placeholders for functionality.